Client/Frontend/Contextual Hint/ContextualHintCopyProvider.swift

```diff
@@ -0,0 +1,103 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/// This specifies the public facing interface any consumers will interact with.
+protocol ContextualHintCopyProviderProtocol {
+    func getCopyFor(_ copyType: ContextualHintCopyType, of hint: ContextualHintType) -> String
+}
+
+/// `ContextualHintCopyProvider` exists to provide the requested description or action strings back
+/// for the specified `ContextualHintType`.
+struct ContextualHintCopyProvider: ContextualHintCopyProviderProtocol, FeatureFlaggable {
+    typealias CFRStrings = String.ContextualHints
+
+    /// Arrow direction infuences toolbar copy, so it exists here.
+    var arrowDirection: UIPopoverArrowDirection?
+
+    init(arrowDirecton: UIPopoverArrowDirection? = nil) {
+        self.arrowDirection = arrowDirecton
+    }
+
+    // MARK: - Public interface
+
+    /// Returns the copy for the requested part of the CFR (`ContextualHintCopyType`) for the specified
+    /// hint.
+    ///
+    /// - Parameters:
+    ///   - copyType: The requested part of the CFR copy to return.
+    ///   - hint: The `ContextualHintType` for which a consumer wants copy for.
+    /// - Returns: The requested copy.
+    func getCopyFor(_ copyType: ContextualHintCopyType, of hint: ContextualHintType) -> String {
+        var copyToReturn: String
+
+        switch copyType {
+        case .action:
+            copyToReturn = getActionCopyFor(hint)
+        case .description:
+            copyToReturn = getDescriptionCopyFor(hint)
+        }
+
+        return copyToReturn
+    }
+
+    // MARK: - Private helpers
+
+    private func getDescriptionCopyFor(_ hint: ContextualHintType) -> String {
+        var descriptionCopy = ""
```

Comment:
nit: Preference for an empty string should be `String()`, rather than an empty string literal. I don't think we implemented that Swiftlint rule, but I think its on there and will be proposed at some point. Not a big deal, however. lol

Reply:
Is there some reasoning for preferring one over the other? If it's just a consistency thing I think you'll get some push back on `String()` being the default because `""` is so ubiquitous.

Reply:
it's a nit, so I honestly don't mind either way. Actually, reading on it, apparently it's the exact same instruction set in assemly so there's literally no difference. Well, TIL! :D